internal/provider/group_resource.go

```diff
@@ -0,0 +1,350 @@
+	}
+	return diags
+}
+
+func (r GroupResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data GroupResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(r.CreateGroup(&data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r GroupResource) DeleteGroup(data GroupResourceModelInterface) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	resp, _, err := r.client.doRequest(http.MethodDelete, data.GetURL(), nil)
+	if err != nil {
+		diags.AddError("Body JSON Marshal Error", err.Error())
+		return diags
+	}
+	if resp == nil {
+		diags.AddError("Http response Error", "no http response from server")
+		return diags
+	}
+	if resp.StatusCode != http.StatusNoContent {
+		diags.AddError("Unexpected Http Status code",
+			fmt.Sprintf("expected (%d) got (%s)", http.StatusNoContent, resp.Status))
+		return diags
+	}
+	return diags
+}
+
+func (r GroupResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data GroupResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	resp.Diagnostics.Append(r.DeleteGroup(&data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r GroupResource) UpdateGroup(data GroupResourceModelInterface) diag.Diagnostics {
+	var diags diag.Diagnostics
+	var req_data io.Reader = nil
+	req_body, diagCreateReq := data.CreateRequestBody()
+	diags.Append(diagCreateReq...)
+	if diags.HasError() {
+		return diags
+	}
+	if req_body != nil {
+		req_data = bytes.NewReader(req_body)
+	}
+	resp, body, err := r.client.doRequest(http.MethodPut, data.GetURL(), req_data)
+
+	if err != nil {
+		diags.AddError("Body JSON Marshal Error", err.Error())
+		return diags
+	}
+	if resp == nil {
+		diags.AddError("Http response Error", "no http response from server")
+		return diags
+	}
+	if resp.StatusCode != http.StatusOK {
+		diags.AddError("Unexpected Http Status code",
+			fmt.Sprintf("expected (%d) got (%s)", http.StatusOK, resp.Status))
+		return diags
+	}
+	err = data.ParseHttpResponse(body)
+	if err != nil {
+		diags.AddError("error while parsing the json response: ", err.Error())
+		return diags
+	}
+	return diags
+}
+
+func (r GroupResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data GroupResourceModel
+	var data_with_URL GroupResourceModel
+
+	// Read Terraform plan and state data into the model
+	// The URL is generated once the group is created. To update the correct group, we retrieve the state data and append the URL from the state data to the plan data.
```

Comment:
I don't think this is necessary. With the `UseStateForUnknown` planmodifier, the plan should include the URL from state.